actions/FirefallClient.js

```diff
@@ -17,37 +18,53 @@ class FirefallClient {
     this.apiKey = apiKey;
     this.org = org;
     this.accessToken = accessToken;
+    this.logger = Core.Logger('FirefallClient', { level: 'info' });
   }
 
   async completion(prompt, temperature = 0.0, model = 'gpt-4') {
-    return wretchRetry(`${this.endpoint}/v1/completions`)
-      .headers({
-        'x-gw-ims-org-id': this.org,
-        'x-api-key': this.apiKey,
-        Authorization: `Bearer ${this.accessToken}`,
-        'Content-Type': 'application/json',
-      })
-      .post({
-        dialogue: {
-          question: prompt,
-        },
-        llm_metadata: {
-          llm_type: 'azure_chat_openai',
-          model_name: model,
-          temperature,
-          max_tokens: 4096,
-          top_p: 1,
-          frequency_penalty: 0,
-          presence_penalty: 0,
-          n: 1,
-        },
-      })
-      .json();
+    const startTime = Date.now();
+
+    try {
+      const response = await wretchRetry(`${this.endpoint}/v1/completions`)
+        .headers({
+          'x-gw-ims-org-id': this.org,
+          'x-api-key': this.apiKey,
+          Authorization: `Bearer ${this.accessToken}`,
+          'Content-Type': 'application/json',
+        })
+        .post({
+          dialogue: {
+            question: prompt,
+          },
+          llm_metadata: {
+            llm_type: 'azure_chat_openai',
+            model_name: model,
+            temperature,
+            max_tokens: 4096,
+            top_p: 1,
+            frequency_penalty: 0,
+            presence_penalty: 0,
+            n: 1,
+          },
+        })
+        .json();
+
+      const endTime = Date.now();
+      const requestTime = ((endTime - startTime) / 1000).toFixed(2);
+      this.logger.info(`Generate request completed in ${requestTime} s`);
+
+      return response;
+    } catch (error) {
+      this.logger.error('Failed generate request:', error);
```

Comment:
It appears related to our contract with the support team. Can you add proper error codes or are they automatically added? @roribio could you pls review?